Documentation/design-docs/arm64-intrinsics.md

```diff
@@ -1,217 +1,269 @@
+  + Priority will be given to intrinsics which are equivalent/similar to those actively used in libraries for other platforms
+  + Priority will be given to intrinsics which have already been implemented for other platforms
+
+### API review
+
+Intrinsics will extend the API of CoreCLR.  They will need to follow standard API review practices.
+
+#### API review of a new intrinsic `static class`es
+
+Review will be facilitated by GitHub Issues requests.
 
-#### Addition of new intrinsics namespaces
-When an intrinsic namespace is introduced code will likely be implemented on different platforms at different times.
+API reviews will be completed within the CoreFX project.
 
-It is explicitly OK for ARM64 to implement a namespace which may be implemented on `ARM32` w/o `ARM32` implementation.
-The opposite is also true.
+A separate GitHub Issue will be created for the review of each intrinsic `static class`.  This allows design and review team to
+review separately.  This allows reviw complexity to be kept manageable.  The O(N^2) nature of the review process will be kept to 
+reasonable levels and iterations will be finite.
 
-Any platform which does not implement a namespace, must:
+Every effort will be made to completely elaborate all the methods of the intrinsic `static class` for each review.  This will 
+help minimize reopening the same classes for review.
 
-+ Return `false` for `IsSupported`
-+ `throw` on any unimplemented function
+Implementation will be kept separate from from API review.
 
-#### Partial implementation of intrinsic namespace
+### Partial implementation of intrinsic `static class`
 
-+ It is preferred that `IsSupported` represents the state of an entire namespace
-+ It is certainly required at time of a release, `IsSupported` represents the state of an entire namespace
-+ Allow for writing tests and implementing code, `IsSupported == false` may be used when namespace is partially implemented
++ `IsSupported` must represents the state of an entire intrinsic `static class`
++ Once API review is complete and approved, it is acceptable to implement approved methods in any order provided tests
 
-#### Addition of new intrinsics to existing namespace
-When an intrinsic namespace is introduced, a best guess will be made about the set of useful intrinsics.
-However, the intrinsics will be added as needed or requested for various optimization efforts.
+### Addition of new intrinsics methods to an existing `static class`
 
-+ The set of intrinsics instructions supported by a namespace should only be allowed
-if all platforms reporting `IsSupported` are supported (at least at time of release)
-+ It may be preferable to add new namespaces in cases where this creates significant burden
+Since each `ARM64` intrinsic `static class` only supports a single platform, adding new instructions is not
+inherently problematic.  New methods can be added without undue complexity.
+
+The primary problem would be the API review process
```

Comment:
I believe there is also concern over the linked runtime dependency. CC. @CarolEidt 

It may be problematic if a class of intrinsics (`X`) is published with v2.1 of the runtime and a new method is added in v2.2 of the runtime, as the `IsSupported` flag would return true for the class on v2.1, but the underlying method may not actually be implemented.

Reply:
I will drop this.  There is no need for this.

Reply:
Done